internal/services/monitor/monitor_alert_prometheus_rule_group_resource_test.go

```diff
@@ -0,0 +1,350 @@
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r AlertPrometheusRuleGroupTestResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-rg-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_monitor_action_group" "test" {
+  name                = "acctestActionGroup-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  short_name          = "acctestag"
+}
+
+resource "azurerm_monitor_workspace" "test" {
+  name                = "acctest-amw-%[1]d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = "%[2]s"
+}
+
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r AlertPrometheusRuleGroupTestResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_monitor_alert_prometheus_rule_group" "test" {
+  name                = "acctest-amprg-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = "%s"
+  scopes              = [azurerm_monitor_workspace.test.id]
+  rule {
+    expression = "histogram_quantile(0.99, sum(rate(jobs_duration_seconds_bucket{service=\" billing-processing \"}[5m])) by (job_type))"
+    record     = "job_type:billing_jobs_duration_seconds:99p5m"
+    labels = {
+      team = "prod"
+    }
+  }
+}
+`, template, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r AlertPrometheusRuleGroupTestResource) requiresImport(data acceptance.TestData) string {
+	config := r.basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_monitor_alert_prometheus_rule_group" "import" {
+  name                = azurerm_monitor_alert_prometheus_rule_group.test.name
+  resource_group_name = azurerm_monitor_alert_prometheus_rule_group.test.resource_group_name
+  location            = azurerm_monitor_alert_prometheus_rule_group.test.location
+  scopes              = azurerm_monitor_alert_prometheus_rule_group.test.scopes
+
+  rule {
+    expression = "histogram_quantile(0.99, sum(rate(jobs_duration_seconds_bucket{service=\" billing-processing \"}[5m])) by (job_type))"
+    record     = "job_type:billing_jobs_duration_seconds:99p5m"
+    labels = {
+      team = "prod"
+    }
+  }
+}
+`, config)
+}
+
+func (r AlertPrometheusRuleGroupTestResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%[2]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%[2]d"
+  kubernetes_version  = "1.25.5"
```

Comment:
Just passing by as I was researching `azurerm_monitor_workspace` and AKS Prometheus metrics, but should this `kubernetes_version` maybe be based on https://github.com/hashicorp/terraform-provider-azurerm/blob/35b106b252848638425164da5b73b1d6e824ffca/internal/services/containers/kubernetes_cluster_resource_test.go#L22 ?

Reply:
> Just passing by as I was researching `azurerm_monitor_workspace` and AKS Prometheus metrics, but should this `kubernetes_version` maybe be based on `currentKubernetesVersion` ?
> 
> https://github.com/hashicorp/terraform-provider-azurerm/blob/35b106b252848638425164da5b73b1d6e824ffca/internal/services/containers/kubernetes_cluster_resource_test.go#L22
> 
> Thanks a lot anyway for creating this resource 👍🏽

Yes, `kubernetes_version` is value of `currentKubernetesVersion`. Thanks for your reminder. Per the [description ](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#kubernetes_version)of `kubernetes_version`, _`If not specified, the latest recommended version will be used at provisioning time`_, so in all tests for resource  `azurerm_monitor_alert_prometheus_rule_group` I removed the specified `kubernetes_version`  to reduce dependencies. Please let me know if you have other suggestions.